### PR TITLE
added uiButton + uiSlider bindings

### PR DESCRIPTION
**Pull request by superzazu**
I added the uiButton and uiSlider bindings. I also implemented the "setText"/"getText" in uiLabel.

Feel free to comment if anything is not as you would like, I'm still fairly new to contributing code :-)

### CODE REVIEW

File: examples/button.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+from pylibui.core import App
+from pylibui.controls import Window, Button
+
+
+class MyWindow(Window):
+
+    def onClose(self, data):
+        super().onClose(data)
+        app.stop()
+
```
Comment (joaoventura) [inline examples/button.py]: Regarding style, I usually have two lines separating classes..

File: examples/button.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class MyButton(Button):
+
+    def onClick(self, data):
+        if self.getText() == 'click me':
+            self.setText('click me again')
+        else:
+            self.setText('click me')
+
```
Comment (joaoventura) [inline examples/button.py]: And separating the class from the following code..

File: pylibui/controls/__init__.py [hunk 3]
```diff
@@ -4,6 +4,10 @@
...
 
 from .box import Box, HorizontalBox, VerticalBox
+from .button import Button
 from .control import Control
+from .checkbox import Checkbox
 from .label import Label
-from .window import Window
+from .progressbar import ProgressBar
+from .slider import Slider
+from .window import Window
```
Comment (joaoventura) [inline pylibui/controls/__init__.py]: It's important to add a newline after the last piece of code. Some IDE's can be set to include it automatically.

File: examples/progressbar.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+
+class MyWindow(Window):
+
+    def onClose(self, data):
+        super().onClose(data)
+        app.stop()
+
+class MyProgressBar(ProgressBar):
+    pass
```
Comment (joaoventura) [inline examples/progressbar.py]: If you are not defining a method with a listener, you don't need to subclass ProgressBar and can use it directly.

File: examples/checkbox.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+from pylibui.core import App
+from pylibui.controls import Window, Checkbox
+
+
+class MyWindow(Window):
+
+    def onClose(self, data):
+        super().onClose(data)
+        app.stop()
+
```
Comment (joaoventura) [inline examples/checkbox.py]: Same thing regarding two empty lines here..

File: examples/checkbox.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+class MyCheckbox(Checkbox):
+    
+    def onToggled(self, data):
+        print('checkbox toggled!')
+
```
Comment (joaoventura) [inline examples/checkbox.py]: And here..

File: examples/progressbar.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+from pylibui.core import App
+from pylibui.controls import Window, ProgressBar
+
+
+class MyWindow(Window):
+
+    def onClose(self, data):
+        super().onClose(data)
+        app.stop()
+
```
Comment (joaoventura) [inline examples/progressbar.py]: Same thing regarding two empty lines here..

File: examples/slider.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+from pylibui.core import App
+from pylibui.controls import Window, Slider
+
+
+class MyWindow(Window):
+
+    def onClose(self, data):
+        super().onClose(data)
+        app.stop()
+
```
Comment (joaoventura) [inline examples/slider.py]: Same thing regarding two empty lines here..

File: examples/slider.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+class MySlider(Slider):
+    
+    def onChanged(self, data):
+        print(self.getValue())
+
```
Comment (joaoventura) [inline examples/slider.py]: And here

File: pylibui/controls/button.py [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    def getText(self):
+        """
+        Returns the text of the button
+
+        :return: string
+        """
+        return libui.uiButtonText(self.control)
+
+    def onClick(self, data):
+        pass
```
Comment (joaoventura) [inline pylibui/controls/button.py]: newline needed here as well (can you see the red symbol? in the github website..)

File: pylibui/controls/button.py [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+    def getText(self):
+        """
+        Returns the text of the button
+
+        :return: string
+        """
+        return libui.uiButtonText(self.control)
+
+    def onClick(self, data):
```
Comment (joaoventura) [inline pylibui/controls/button.py]: Docstring missing..

File: pylibui/controls/slider.py [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    def getValue(self):
+        """
+        Returns the value of the slider.
+
+        :return: int
+        """
+        return libui.uiSliderValue(self.control)
+
+    def onChanged(self, data):
+        pass
```
Comment (joaoventura) [inline pylibui/controls/slider.py]: Docstring and newline at the end missing

**Comment by joaoventura** [comment]
@superzazu , your code is very good, there's only some things regarding style which are "missing" but are not problematic at all. I've tested, everything is working ok and I'm going to merge it, but it would be great if you could read my suggestions and do a final commit with the "fixes".. 👍

**Comment by superzazu** [comment]
Sorry for the inconsistencies !
I pushed a commit (a7f661c) which should fix every problem you raised. :-)

**Comment by joaoventura** [comment]
Great, thanks for the contribution! 👍

**Pull request merged**